pkg/vpcmodel/grouping.go

```diff
@@ -612,14 +614,18 @@ func (details *srcDstDetails) explanationEncode(c *VPCConfig) string {
 	if details.crossVpcRouter != nil {
 		encodeComponents = append(encodeComponents, details.crossVpcRouter.UID())
 	}
-	if len(details.actualMergedRules.egressRules) > 0 {
-		encodeComponents = append(encodeComponents, "egress:"+
-			details.actualMergedRules.egressRules.rulesDetailsStr(c, details.filtersRelevant, false))
-	}
-	if len(details.actualMergedRules.ingressRules) > 0 {
-		encodeComponents = append(encodeComponents, "ingress:"+
-			details.actualMergedRules.ingressRules.rulesDetailsStr(c, details.filtersRelevant, true))
-	}
-
+	details.actualMergedRules.egressRules.appendEncodeRules(&encodeComponents, c, details.filtersRelevant,
+		"egress", false)
+	details.actualMergedRules.ingressRules.appendEncodeRules(&encodeComponents, c, details.filtersRelevant,
+		"ingress", true)
 	return strings.Join(encodeComponents, ";")
 }
+
+func (rules *rulesInLayers) appendEncodeRules(encodeComponents *[]string,
+	c *VPCConfig, filtersRelevant map[string]bool, header string, isIngress bool) {
```

Comment:
"who is the receiver" comment

Reply:
Modified to be a parm and not the receiver

Reply:
In 2nd thought, I'm undoing the change. To be consistent, if we decide not to have e.g. *rulesInLayer the receiver is such cases then this should be part of a separate PR since it is across the board.